### PR TITLE
Doc: Simplify the definition of 'soft deprecated'

### PR DESCRIPTION
**Pull request by andresdelfino**
I found the definition of Soft deprecated somewhat complicated, and I wanted to give it a try.

The violation of line max width is intentional, as I was once told it's better for diff purposes, until merge.

----
📚 Documentation preview 📚: https://cpython-previews--124988.org.readthedocs.build/

### CODE REVIEW

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
      The main difference from a normal deprecation
      is that soft deprecation does not plan on removing the API.
```

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
      Another difference is that soft deprecated APIs don't emit warnings.
```

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
      A soft deprecated API should not
      be used in new code, but it is safe for already existing code to use it.
      The API remains documented and tested, but will not be
      enhanced further.
```

**Review by AA-Turner — COMMENTED** [review]
This is a good start towards a rewrite, thank you. I've suggested some further improvements inline:

> The violation of line max width is intentional, as I once told it's better for diff purposes, at least in the PR stage.

This is fine but we should re-wrap before merge.

A

**Comment by andresdelfino** [comment]
@AA-Turner Thanks for the review! I have addressed all comments. Sorry if you already received a notification :sweat_smile:

File: Doc/glossary.rst
Comment (skirpichev) [inline Doc/glossary.rst]:
```suggestion
      A soft deprecated API should not be used in new code, but it is
      safe for already existing code to use it.  The API remains
      documented and tested, but will not be enhanced further.
```

File: Doc/glossary.rst
Comment (skirpichev) [inline Doc/glossary.rst]:
```suggestion
      The main difference from a normal deprecation is that soft
      deprecation does not plan on removing the API.
```

**Review by skirpichev — APPROVED** [review]
LGTM, modulo two nitpicks (run fill-region in emacs, feel free to ignore).

PS: @andresdelfino, please avoid "Update branch" button, unless you are fixing merge conflicts or something like that (https://devguide.python.org/getting-started/pull-request-lifecycle/#update-branch-button).

File: Doc/glossary.rst
Comment (willingc) [inline Doc/glossary.rst]: We could simplify and remove the last added sentence too.

```suggestion
      Soft deprecation, unlike normal deprecation, does not plan on removing the API and
      will not emit warnings.
```

**Review by willingc — COMMENTED** [review]
Thanks!

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
```

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
      A soft deprecated API should not be used in new code, 
      but it is safe for already existing code to use it.
      The API remains documented and tested, but will not be enhanced further.
```

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
      Soft deprecation, unlike normal deprecation, does not plan on removing the API 
      and will not emit warnings.
```

**Review by AA-Turner — APPROVED** [review]
Remove the final sentence per Carol and re-wrap.

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
      A soft deprecated API should not be used in new code,
```

File: Doc/glossary.rst
Comment (AA-Turner) [inline Doc/glossary.rst]:
```suggestion
      Soft deprecation, unlike normal deprecation, does not plan on removing the API
```

**Pull request merged**
